sdk/data/aztables/README.md

```diff
@@ -40,7 +40,7 @@ Once you have the account URL, it can be used to create the service client:
 ```golang
 cred, err := aztables.NewSharedKeyCredential("myAccountName", "myAccountKey")
 handle(err)
-serviceClient, err := aztables.NewServiceClient("https://<myAccountName>.table.core.windows.net/", cred, nil)
+serviceClient, err := aztables.NewServiceClientWithSharedKeyCredential("https://<myAccountName>.table.core.windows.net/", cred, nil)
```

Comment:
@chlowell : Do we want to clean up the "shared key credential" story? Specifically, do we want a NameKeyCredential in azcore which customers use here which would rename this func to NewServiceWithNamedKeyCredential.

And then other client libs can use azcore's NamedKeyCredential (with each lib providing its own policy for how it should work for tht service).